### PR TITLE
Feat : Refactor

### PR DESCRIPTION
**Pull request by srejitk**
This feature request adds the following features:

1. Responsiveness
2. Refactor
3. Error Page
4. Style Updates
5. Bug Fixes

### CODE REVIEW

File: social-circus/src/components/EditPostModal/EditPostModal.jsx [hunk 1]
```diff
@@ -60,21 +63,29 @@ export const EditPostModal = ({ openModal, setOpenModal, edit }) => {
...
+    setForm({
+      ...form,
+      imageUrl: imagePath,
+      username: user?.username,
+      avatar: user?.avatar,
+    });
     if (form.content !== "") {
       edit
         ? await EditPost(form, dispatch, post?.id)
         : await AddPost(form, dispatch);
```
Comment (rahulcs754) [inline social-circus/src/components/EditPostModal/EditPostModal.jsx:75]: nice use to ternary operator

File: social-circus/src/routes/Home/Home.jsx [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
   const { user, token } = useSelector((state) => state.auth);
-  let homeposts = [];
+  const followingUsers = user?.following;
+  const homeposts = posts?.filter(
+    (post) => followingUsers?.includes(post?.uid) || post?.uid === token
+  );
+
+  const mediaposts = [...homeposts].filter((post) => post?.imageUrl !== "");
+
+  console.log(posts);
```
Comment (rahulcs754) [inline social-circus/src/routes/Home/Home.jsx]: remove console.log before upload

File: social-circus/src/routes/Home/Home.jsx [hunk 3]
```diff
@@ -51,10 +57,10 @@ export const Home = () => {
                       : "text-blue-100 hover:bg-white/[0.12] hover:text-white"
                   }`}
                 >
-                  Latest
+                  Old School
                 </button>
               )}
-            </Tab>
+            </Tab>{" "}
```
Comment (rahulcs754) [inline social-circus/src/routes/Home/Home.jsx:61]: remove empty space

**Review by rahulcs754 — APPROVED** [review]
Everything looks good.  Code looks nice and clean.

**Review by sadathshariff — APPROVED** [review]
Looks good, no changes required

File: social-circus/src/redux/actions/uploadImageActions.jsx [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { storage } from "../../firebase";
 export const UploadImage = async (path, file) => {
   try {
     const imageRef = ref(storage, path);
-    const response = await uploadBytesResumable(imageRef, file);
+    const response = file && (await uploadBytesResumable(imageRef, file));
     const pathName = response?.ref?.toString();
```
Comment (shubhamkumar648) [inline social-circus/src/redux/actions/uploadImageActions.jsx:9]: perfect use of nullish operator

**Review by shubhamkumar648 — APPROVED** [review]
- well written  structure and consistency code 
- good use of bullish operator 
- good to merge

**Pull request merged**